File: 总题库/DivingBoard.py
class Solution:
    def divingBoard(self, shorter: int, longer: int, k: int) -> List[int]:
        res = []
        if k == 0:
            return res
        tempLen = shorter * k
        res.append(tempLen)
        if shorter == longer:
            return res

        for i in range(1, k + 1):
            res.append(tempLen + i * (longer - shorter))
        return res

File: 总题库/DivingBoard.cpp
class Solution {
public:
    vector<int> divingBoard(int shorter, int longer, int k) {
        vector<int> res;
        if (k == 0) return res;
        int tempLen = shorter * k;
        res.push_back(tempLen);
        if (shorter == longer) return res;
        for (int i = 1; i <= k; ++i) {
            res.push_back(tempLen + i * (longer - shorter));
        }         
        return res;
    }
};